mods/sethome/init.lua

```diff
@@ -24,42 +25,69 @@ loadhomes()
 
 minetest.register_privilege("home", "Can use /sethome and /home")
 
-local changed = false
+function sethome.go_home(name)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		-- just a check to prevent the server crashing
+		return false
+	end
+	if homepos[name] then
+		player:setpos(homepos[name])
+		return true
+	else
+		return false
+	end
+end
 
 minetest.register_chatcommand("home", {
-    description = "Teleport you to your home point",
-    privs = {home=true},
-    func = function(name)
-        local player = minetest.get_player_by_name(name)
-        if player == nil then
-            -- just a check to prevent the server crashing
-            return false
-        end
-        if homepos[player:get_player_name()] then
-            player:setpos(homepos[player:get_player_name()])
-            minetest.chat_send_player(name, "Teleported to home!")
-        else
-            minetest.chat_send_player(name, "Set a home using /sethome")
-        end
-    end,
+	description = "Teleport you to your home point",
+	privs = {home=true},
+	func = function(name)
+		local res = sethome.go_home(name)
+		if res then
+			return true, "Teleported to home!"
+		else
+			return false, "Set a home using /sethome"
+		end
+	end,
 })
 
+function sethome.set_home(name, pos)
+	local player = minetest.get_player_by_name(name)
+	if not player then
```

Comment:
I see how it could be a good idea, but that's just too harsh to code. I can't process messages both in and out of the function, otherwise I could just put them back as chatcommand callbacks.
Also, the reason I do not always require pos to be a table, or even to be passed, is that when used in the chatcommand, `sethome.set_home` will only have to send the player's name, and let the function check whether or not that's a player's name. If I put pos as a mandatory parameter, then the chatcommand callback should have to check if 'name' is the name of a player (doing part of the job of sethome's set_home) and then give the player's position to sethome.set_home. But then, other mods using this function would have to also check their 'name' parameters, and I'm not okay with having to let mods check such details before using another mods' function. Then, if I let the check in sethome.set_home, that would mean I would have to double check in the chatcommand?

Or I could simply use your `assert` line and let the case when pos is nil be accepted.